File: go/general/897. 递增顺序查找树.go
package main

var res []int

func increasingBST(root *TreeNode) *TreeNode {
	res = make([]int, 0)
	dfs(root)
	node := &TreeNode{
		Val: res[0],
	}
	start := node
	for _, v := range res[1:] {
		next := &TreeNode{Val: v}
		node.Right = next
		node = node.Right
	}
	return start
}

func dfs(root *TreeNode) {
	if root != nil {
		dfs(root.Left)
		res = append(res, root.Val)
		dfs(root.Right)
	}
}

File: python/树/897. 递增顺序查找树.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def increasingBST(self, root):
        res = []
        def inorder(node):
            if node:
                inorder(node.left)
                res.append(node.val)
                inorder(node.right)
        inorder(root)
        start = cur = TreeNode(res[0])
        for i in res[1:]:
            cur.right = TreeNode(i)
            cur = cur.right
        return start